pkgs/servers/web-apps/kavita/default.nix

```diff
@@ -0,0 +1,63 @@
+{ lib
+, fetchFromGitHub
+, buildDotnetModule
+, buildNpmPackage
+, dotnetCorePackages
+, nixosTests
+}:
+
+buildDotnetModule rec {
+  pname = "kavita";
+  version = "0.7.1.4";
+
+  src = fetchFromGitHub {
+    owner = "kareadita";
+    repo = "kavita";
+    rev = "v${version}";
+    hash = "sha256-jNhiwyz6iVSLlvMNjI689TwQYuEvTJ+QaPvvDQ4UOwc=";
+  };
+
+  patches = [
+    # So that we can set the webroot
+    ./change-webroot.diff
+  ];
+
+  frontend =  buildNpmPackage {
+    pname = "kavita-web";
+    inherit version src;
+
+    sourceRoot = "source/UI/Web";
+
+    npmBuildScript = "prod";
+    npmFlags = [ "--legacy-peer-deps" ];
+    npmRebuildFlags = [ "--ignore-scripts" ]; # Prevent playwright from trying to install browsers
+    npmDepsHash = "sha256-w0CuTPyCQyAxULvqd6+GiZaPlO8fh4xLmbEnGA47pL8=";
+  };
+
+  postPatch = ''
+    sed -i "s:@WEB_ROOT@:$out/lib/kavita-frontend:" API/Controllers/FallbackController.cs API/Startup.cs
```

Comment:
There is a specific function for those @@-substitutions, namely `substituteAll`. 
Look at the example:

https://github.com/NixOS/nixpkgs/blob/6027cc8c1c4567d8e6d02c5ac1e4b8ff1193c08d/pkgs/tools/system/wsysmon/default.nix#L26-L29

Reply:
Wow that's pretty nice. There really is a function for everything

Reply:
There seems to be an issue with this approach.

It apparently can't access `$out` or `$frontend`, so the only way is to use `${frontend}`, but then the frontend attribute won't be overridable (which is something that seems worth keeping). Any ideas?

Reply:
Oh apparently I can use the `finalAttrs` approach instead.

Reply:
The solution I came up with is using mkDerivation with `backend` and `frontend` as attrs, let me know what you think.

Reply:
Unfortunately not all `build<ecosystem>Package`ers were converted to accept the `(finalAttrs: {...})` construction.